File: Data Preprocessing/Python/data_preprocessing.py
# -*- coding: utf-8 -*-
# Importing the libaries 
import numpy as np
import matplotlib.pyplot as mathplot
import pandas as pd

#Importing the dataset
dataset = pd.read_csv('Data.csv')

# defining matrix of features include all lines and  all columns except the last
X = dataset.iloc[:, :-1].values

#defining the dependent variable vector
y = dataset.iloc[:, 3].values

#handling missing data  using Imputer class from sklearn to completing missing values
from sklearn.preprocessing import Imputer

# In this case the essential paramenters we need to achieve the imputatation transformation are the missing values 
# which is show as nan in our current dataset, we will make use of the mean strategy and to do that our axis will 
# be the columns of feature
imputer = Imputer(missing_values = 'NaN', strategy = 'mean', axis = 0)

# now we need to fit  in our matrix more precisly in the columns were we have the missing values which is 1 and 2
imputer = imputer.fit(X[:,1:3])

# after fit we can replace the missing data by the mean of the columns using transform
X[:,1:3] = imputer.transform(X[:, 1:3])

File: Data Preprocessing/R/data_preprocessing.R
#Data Preprocessing 

#Importing dataset
dataset = read.csv('Data.csv')
 
#Handling missing data 
# we will take the column of Age of the dataset and we use a ifelse condition to check for missing values 
# if the condition is true we get the mean of the collumn to add it in the missing value else it should returns 
# the value of the observation
dataset$Age = ifelse(is.na(dataset$Age),
                     ave(dataset$Age, FUN = function(replaceMissingAgeValues) mean(replaceMissingAgeValues,na.rm = TRUE)),
                     dataset$Age) 

# we applied the same strategy to replace the missing salary values from the dataset
dataset$Salary = ifelse(is.na(dataset$Salary),
                        ave(dataset$Salary, FUN = function(replaceMissingSalaryValues) mean(replaceMissingSalaryValues, na.rm = TRUE)),
                        dataset$Salary)
                        )